kie-dmn-feel/src/main/java/org/kie/dmn/feel/parser/feel11/ParserHelper.java

```diff
@@ -69,18 +80,28 @@ public void recoverScope() {
     }
 
     public void recoverScope( String name ) {
+        LOG.trace("[{}] recoverScope( name: {}) with currentScope: {}", this.currentScope.getName(), name, currentScope);
         Scope s = this.currentScope.getChildScopes().get( name );
         if( s != null ) {
             currentScope = s;
         } else {
             pushScope();
```

Comment:
This is easier to explain over bluejeans. Ping me when you have time to discuss it.